processing/src/main/java/org/apache/druid/segment/column/CachingStringDictionaryEncodedColumn.java

```diff
@@ -54,7 +54,7 @@
 /**
  *
  */
-public class StringDictionaryEncodedColumn implements DictionaryEncodedColumn<String>
+public class CachingStringDictionaryEncodedColumn implements DictionaryEncodedColumn<String>
```

Comment:
oops, I was frustrated that I couldn't re-use it for my optimized nested scalar string column because it requires a `CachingIndexed<String>` which itself requires a `GenericIndexed<String>`, and all I have is either a `GenericIndexed<ByteBufffer>` or a `FrontCodedIndexed` depending on string encoding config

Will change the name back